container/navigation.go

```diff
@@ -0,0 +1,118 @@
+package container
+
+import (
+	"fyne.io/fyne/v2"
+	"fyne.io/fyne/v2/theme"
+	"fyne.io/fyne/v2/widget"
+)
+
+// Navigation container is used to provide your application with a control bar and an area for content objects.
```

Comment:
If the idea is for this to be the final API, it should probably be called StackNavigation, to distinguish it from other navigation schemes, such as a browser-like queue of past pages that can be navigated back and forward. Unless the idea is to make this single component support multiple navigation schemes in the future

Reply:
But couldn't you build a browser back/forward with this too? It could be extended to have forward navigation without needing Stack vs not-stack navigation.
Conceptually it feels like the browser is a stack as well?

Reply:
Is that really the best option? Do we know that all the other navigators will have the same API?

With more and more types in the container package that actually aren't simple containers but technically widgets, I wonder if a "navigation" package would make sense? Tabs would fit in quite well there as well, I'd think. Just an idea to bring up before we commit to anything :)

Reply:
I'm not sure that tabs fit a mental model of navigation - you're not "going" anywhere...
Container has diversified but the general concept of "you put other widgets in there" mostly holds.

Reply:
I don't really think you could build a browser-like navigation with this component without just forking it and adding a different API - the browser navigation model is only a bit "stack-like", but it's really more of a list with a pointer that can advance back and forward, and when you push a new page, it deletes the whole "tail" of the history after the current pointer position. (ie back, back, back, new page would remove the last 3 items in the old history)

Reply:
I'm now considering to change the logic to support an optional "Next" button.

Reply:
Does that mean the naming works now? If it needs a more explicit name then OK but I'm not convinced that "Stack" or "Simple" add anything meaningful to a developer's understanding of what it is.

Reply:
I think the naming works if we plan to add (at least eventually) the forward navigation as well. I wonder if users may also confuse this with a "routing" system though, which would be more complex still.

Reply:
The forward navigation is now in the PR.

Reply:
I'm happy with this now. If a routing system is to be added that would not be presented as a navigation container in this sense - more likely it would have to be a parser of path through a UI which then populates the various containers / tabs along the way to that content. 

Maybe that's overly complex but I'm happy that routing is not a widget/container concern so there is no name conflict.